sdk/dapp-kit/src/components/AutoConnectWallet.tsx

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ReactNode } from 'react';
+import { useEffect, useState } from 'react';
+import { useConnectWallet } from '../hooks/wallet/useConnectWallet.js';
+import { useWalletContext } from '../hooks/wallet/useWalletContext.js';
+import type { StorageAdapter } from '../utils/storageAdapters.js';
+import type { WalletAccount, WalletWithRequiredFeatures } from '@mysten/wallet-standard';
+
+interface AutoConnectWalletProps {
+	storageKey: string;
+	storageAdapter: StorageAdapter;
+	children: ReactNode;
+}
+
+export function AutoConnectWallet({
+	storageKey,
+	storageAdapter,
+	children,
+}: AutoConnectWalletProps) {
+	const { wallets, currentWallet: wallet, currentAccount: account } = useWalletContext();
+	const [previousWallet, setPreviousWallet] = useState<WalletWithRequiredFeatures | null>(null);
+	const [currentWallet, setCurrentWallet] = useState<WalletWithRequiredFeatures | null>(wallet);
+	const [previousAccount, setPreviousAccount] = useState<WalletAccount | null>(null);
+	const [currentAccount, setCurrentAccount] = useState<WalletAccount | null>(account);
+	const { mutate: connectWallet } = useConnectWallet();
+
+	// Instead of abstracting the previous wallet and account state logic into a generic
+	// usePrevious hook, we'll write this plainly to make it clear that this code depends
+	// on triggering shallow re-renders by updating state in the render code.
+	if (wallet !== currentWallet) {
```

Comment:
not sure if it matters, but should we check if currentWallet === previousWallet as well? maybe setState already does this